tests/provider: MQ test failing in GovCloud

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

These tests fails on GovCloud:

```
TestAccDataSourceAWSMqBroker_basic
```

With this error: `no such host`.

### New or Affected Resource(s)

<!--- Please list the new or affected resources and data sources. --->

* aws_mq_broker

### Terraform Configuration Files

Example config:

```hcl
resource "aws_mq_broker" "acctest" {
  auto_minor_version_upgrade = true
  apply_immediately          = true
  broker_name                = "yakbroker"

  configuration {
    id       = aws_mq_configuration.acctest.id
    revision = aws_mq_configuration.acctest.latest_revision
  }

  deployment_mode    = "ACTIVE_STANDBY_MULTI_AZ"
  engine_type        = "ActiveMQ"
  engine_version     = "5.15.0"
  host_instance_type = "mq.t2.micro"

  maintenance_window_start_time {
    day_of_week = "TUESDAY"
    time_of_day = "02:00"
    time_zone   = "CET"
  }

  publicly_accessible = true
  security_groups     = aws_security_group.acctest[*].id
  subnet_ids          = aws_subnet.acctest[*].id

  user {
    username = "Ender"
    password = "AndrewWiggin"
  }

  user {
    username       = "Petra"
    password       = "PetraArkanian"
    console_access = true
    groups         = ["dragon", "salamander", "leopard"]
  }

  depends_on = [aws_internet_gateway.acctest]
}
```

### Debug Output

```
TestAccDataSourceAWSMqBroker_basic: data_source_aws_mq_broker_test.go:20: Step 1/2 error: Error running apply: 2020/11/08 10:03:29 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0
        
        Error: RequestError: send request failed
        caused by: Post "https://mq.us-gov-west-1.amazonaws.com/v1/configurations": dial tcp: lookup mq.us-gov-west-1.amazonaws.com on 192.168.22.2:53: no such host

--- FAIL: TestAccDataSourceAWSMqBroker_basic (40.97s)
```

### References

* #0000

--- a/aws/data_source_aws_mq_broker_test.go
+++ b/aws/data_source_aws_mq_broker_test.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/aws/aws-sdk-go/service/mq"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
@@ -18,7 +19,7 @@ func TestAccDataSourceAWSMqBroker_basic(t *testing.T) {
 	resourceName := "aws_mq_broker.acctest"
 
 	resource.ParallelTest(t, resource.TestCase{
-		PreCheck:  func() { testAccPreCheck(t) },
+		PreCheck:  func() { testAccPreCheck(t); testAccPartitionHasServicePreCheck(mq.EndpointsID, t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{

